engine/servermaster/server.go

```diff
@@ -689,6 +689,15 @@ func (s *Server) createHTTPServer() (*http.Server, error) {
 			MarshalOptions:   protojson.MarshalOptions{UseProtoNames: true, EmitUnpopulated: true},
 			UnmarshalOptions: protojson.UnmarshalOptions{},
 		}),
+		runtime.WithErrorHandler(func(ctx context.Context, mux *runtime.ServeMux,
```

Comment:
Add some unit tests for differnet scenaries please.

Reply:
Addressed in e9a52d35c0463dcc6323d115f1e402e6209b7027.